mesonbuild/minit.py

```diff
@@ -201,13 +208,12 @@ def add_arguments(parser):
     parser.add_argument("-n", "--name", help="project name. default: name of current directory")
     parser.add_argument("-e", "--executable", help="executable name. default: project name")
     parser.add_argument("-d", "--deps", help="dependencies, comma-separated")
-    parser.add_argument("-l", "--language", choices=['c', 'cpp', 'cs', 'cuda', 'd', 'fortran', 'java', 'rust', 'objc', 'objcpp'],
-                        help="project language. default: autodetected based on source files")
-    parser.add_argument("-b", "--build", help="build after generation", action='store_true')
-    parser.add_argument("--builddir", help="directory for build", default='build')
+    parser.add_argument("-l", "--language", choices=LANG_SUPPORTED, help="project language. default: autodetected based on source files")
+    parser.add_argument("-b", "--build", action='store_true', help="build after generation")
+    parser.add_argument("--builddir", default='build', help="directory for build")
     parser.add_argument("-f", "--force", action="store_true", help="force overwrite of existing files and directories.")
-    parser.add_argument('--type', default='executable', choices=['executable', 'library'])
-    parser.add_argument('--version', default='0.1')
+    parser.add_argument('--type', default='executable', choices=['executable', 'library'], help="project type. default: executable based project")
```

Comment:
if you make this an enum as I suggested you can use the `type` parameter here to have argparse do the conversion for you.